src/hotspot/cpu/ppc/c2_init_ppc.cpp

```diff
@@ -51,4 +51,9 @@ void Compile::pd_compiler2_init() {
     FLAG_SET_DEFAULT(OptimizeFill, false);
   }
 
+  if (OptoScheduling) {
+    // The OptoScheduling information is not maintained in ppd.ad.
+    warning("OptoScheduling is not supported on this CPU.");
+    FLAG_SET_DEFAULT(OptoScheduling, false);
+  }
```

Comment:
Makes sense but better do it in `VM_Version::initialize()` because `Compile::pd_compiler2_init()` is called after initialization of flags has been completed and the setting will not be shown with `PrintFlagsFinal`.
I'd even suggest to move the other flag settings there with this PR.

Reply:
This makes sense. Please see my update. I have also added `guarantee(CodeEntryAlignment >= InteriorEntryAlignment, "");` to `Compile::pd_compiler2_init()` which is there on other platforms.